internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,510 @@
+			}
+		}
+	}
+
+	h.log.Debugf("Got %d releases", len(releases))
+	return releases, nil
+}
+
+func (h *releaseSourcesHandler) getStaticReleaseImages() []*common.ReleaseImage {
+	dbReleaseImages := []*common.ReleaseImage{}
+	for _, releaseImage := range h.staticReleaseImages {
+		dbReleaseImage := common.ReleaseImage{ReleaseImage: *releaseImage}
+		// For backward compatibility with release images that lack the CPUArchitectures field.
+		if dbReleaseImage.CPUArchitectures == nil {
+			dbReleaseImage.CPUArchitectures = []string{*dbReleaseImage.CPUArchitecture}
+		}
+		dbReleaseImage.DBCPUArchitectures = dbReleaseImage.CPUArchitectures
+		dbReleaseImages = append(dbReleaseImages, &dbReleaseImage)
+	}
+
+	return dbReleaseImages
+}
+
+// mergeDBReleaseImages merges static and dynamic release images together such that
+// static release images have precedence in case of conflicts e.g. which release is default
+func (h *releaseSourcesHandler) mergeDBReleaseImages(staticReleaseImages, dynamicReleaseImages []*common.ReleaseImage) []*common.ReleaseImage {
+	mergedReleaseImages := []*common.ReleaseImage{}
+	staticReleaseImagesReferenceSet := map[string]bool{}
+	defaultStaticReleaseExists := false
+
+	for _, staticRelease := range staticReleaseImages {
+		mergedReleaseImages = append(mergedReleaseImages, staticRelease)
+		staticReleaseImagesReferenceSet[*staticRelease.URL] = true
+		if staticRelease.Default {
+			defaultStaticReleaseExists = true
+		}
+	}
+
+	for _, dynamicRelease := range dynamicReleaseImages {
+		if staticReleaseImagesReferenceSet[*dynamicRelease.URL] {
+			continue
+		}
+		if dynamicRelease.Default && defaultStaticReleaseExists {
+			dynamicRelease.Default = false
+		}
+		mergedReleaseImages = append(mergedReleaseImages, dynamicRelease)
+	}
+
+	return mergedReleaseImages
+}
+
+func (h *releaseSourcesHandler) syncReleaseImagesWithErr() error {
+	err := h.validateReleaseSources()
+	if err != nil {
+		return err
+	}
+
+	err = h.validateStaticReleaseImages()
+	if err != nil {
+		return err
+	}
+	dynamicDBReleaseImages, err := h.getDynamicReleaseImages()
+	if err != nil {
+		return err
+	}
+	h.log.Debugf("Found %d dynamic release Images", len(dynamicDBReleaseImages))
+	supportLevels, err := h.getSupportLevels()
+	if err != nil {
+		return err
+	}
+	err = h.setSupportLevels(dynamicDBReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	staticDBReleaseImages := h.getStaticReleaseImages()
+	h.log.Debugf("Found %d static release Images", len(staticDBReleaseImages))
+	err = h.setSupportLevels(staticDBReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	err = h.setDefaultReleaseImage(dynamicDBReleaseImages)
+	if err != nil {
+		return err
+	}
+
+	dynamicDBReleaseImages, err = h.removeReleaseImageDuplicates(dynamicDBReleaseImages)
+	if err != nil {
+		return err
+	}
+
+	dbReleaseImages := h.mergeDBReleaseImages(staticDBReleaseImages, dynamicDBReleaseImages)
+
+	h.log.Debug("Starting SQL transaction")
+	tx := h.db.Begin()
```

Comment:
why not to use same template we use at all other places 
`txSuccess := false
	tx := h.db.Begin()
	defer func() {
		if !txSuccess {
			log.Error("update monitored operator failed")
			tx.Rollback()
		}
		if r := recover(); r != nil {
			log.Error("update monitored operator failed")
			tx.Rollback()
		}
	}()`
no idea though why we are not making it a function to be reusable

Reply:
I think the main issue with this is the `recover`, I had a talk about it with @filanov and @rccrdpccl, we came to a conclusion that hides errors, therefore we should not recover.

Reply:
We can create a function without recovering though

Reply:
all the recovers should already be removed, we replaced the code with using a new format that will fail the transaction and handle the rollback https://github.com/openshift/assisted-service/blob/a8dca1f6d37e0e17b37c93e1c24426fc9f80ff1c/internal/bminventory/inventory.go#L1288 
otherwise this recover will return a `nil, nil` as a replay and the caller will think that everything is ok